src/main/scala/io/opentargets/etl/backend/spark/Helpers.scala

```diff
@@ -133,21 +137,71 @@ object Helpers extends LazyLogging {
     }
   }
 
-  def writeTo(outputConfs: IOResourceConfs, outputs: IOResources)(
-      implicit
-      session: SparkSession): IOResources = {
+  def writeTo(outputConfs: IOResourceConfs, outputs: IOResources)(implicit
+      session: SparkSession
+  ): IOResources = {
 
     logger.info(s"Saving data to '${outputConfs.mkString(", ")}'")
 
-    outputConfs foreach {
-      case (n, c) =>
-        logger.debug(s"saving dataframe '$n' into '${c.path}'")
-        outputs(n).write.format(c.format).save(c.path)
+    outputConfs foreach { case (n, c) =>
+      logger.debug(s"saving dataframe '$n' into '${c.path}'")
+      outputs(n).write.format(c.format).save(c.path)
     }
 
     outputs
   }
 
+  /** generate a set of String with the union of Columns.
+    * Eg, myCols =( a,c,d) and allCols(a,c,d,e,f,h)
+    * return (a,c,d,e,f,h)
+    * @param myCols the list of the Columns in a specific Dataframe
+    * @param allCols the list of Columns to match
+    * @return a sparksession object
+    */
+  def columnExpr(myCols: Set[String], allCols: Set[String]) = {
```

Comment:
@cmalangone when you use sets is better to stick to set operations. those are clearer to follow. by example

```scala
  def columnExpr(myCols: Set[String], allCols: Set[String]) = {
    val inter = (allCols intersect myCols).map(col)
    val differ = (allCols diff myCols).map(lit(null).as(_))

    inter union differ
  }
```
Although if you want to stick with your way you can write this cleaner
```scala
  def columnExpr(myCols: Set[String], allCols: Set[String]) = {
    allCols map {
      case x if myCols.contains(x) => col(x)
      case x =>  lit(null).as(x)
    }
  }
```

Reply:
I changed the code as you suggest. (set operation).

Reply:
@mkarmona  and @JarrodBaker I changed the code as both of you suggested